### DISCUSSION TITLE
Global max player ping

### DISCUSSION THREAD

**Discussion by TropicalShadow** [jeremylvln/Shulker#388] [Ideas]
> display the max player count for the proxy as the max cluster player count.

The online player count dynamically scales to the cluster wide online count, however the max player count is the proxy configured player capacity.

It would ideally be better to somehow poll all the proxies and or sub servers then calculate the total max players based on each server capacity and how many server/proxies available (not draining).

maybe have a config to determine the strategy of calculating max player count. Proxy only, all proxies, sub servers.

- proxy only: default how it works now where it uses the proxy config
- All proxies: calculates the max for each proxy adding them all up
- Sub servers: calculate all the sub servers, adding up the max player capacity for all

**Comment by jeremylvln** [comment]
Hi,

Thanks for pointing out, this makes a lot of sense. Calculating the sum of each proxy capacity and use it for the max players count can work. However, there might be an issue if you are using multiple fleets and want to hide the capacity of one fleet (let's say you have one fleet for the players and one for the staff, you may not want to show the staff reserved slots).

Considering the proxy is already aware of its fleet, we can make them return the max number of slot corresponding to the sum of every proxies in its fleet.

I would prefer to find a correct way to satisfy everyone, having a configuration flag for the strategy may be a bit overkill, however I'm open to the discussion.

Configuration flag or not, fixing the slots count will be not hard.

What do you think?

**Reply by TropicalShadow** [reply]
I think having a configuration flag for the strategy would assist with the problem you have with the staff player count.
let's say we set the strategy to a static number for the staff fleet, but then use the dynamic strategy for the main fleet.

if I am right, the proxy player count doesn't affect the actual proxy capacity, and is just cosmetical?

for now, however just a global max player count is fine, but in the future, I think many people could benefit from the different strategies for calculating the player count.

**Reply by jeremylvln** [reply]
Let's go with that. I'll open an issue with the technical solution and tackle it today (even right now).

**Reply by jeremylvln** [reply]
See #393.

Also, I found out something that needed to be fixed too. See #395.

**Reply by TropicalShadow** [reply]
Much appropriate! Thank you for looking into it.

**Comment by jeremylvln** [comment]
I close the discussion now that dedicated issues were open. Let's continue the discussion on them if needed :)
